### PR TITLE
BP-2872 - Make 'Create invoice after shipment' a general setting

### PR DESCRIPTION
**Pull request by LucianTuriacArnia**

### CODE REVIEW

File: Observer/SalesOrderShipmentAfter.php [hunk 1]
```diff
@@ -131,17 +152,27 @@ public function execute(\Magento\Framework\Event\Observer $observer)
...
             );
             $this->createInvoice($order, $shipment, true);
         }
+
+        if (strpos($paymentMethodCode, 'buckaroo_magento2') !== false
+            && $this->configAccount->getInvoiceHandling() == InvoiceHandlingOptions::SHIPMENT) {
+            if ($paymentMethod->getConfigPaymentAction() == 'authorize') {
+                $this->createInvoice($order, $shipment, true);
+            } else {
+                $this->createInvoice->createInvoiceGeneralSetting($order);
```
Comment (harli91) [inline Observer/SalesOrderShipmentAfter.php:170]: having a method and a property name the same way looks confusing to me

File: view/adminhtml/web/images/svg/pos-nfc.svg
Comment (harli91) [inline view/adminhtml/web/images/svg/pos-nfc.svg:1]: do we need this?

File: view/adminhtml/web/images/svg/riverty.svg
Comment (harli91) [inline view/adminhtml/web/images/svg/riverty.svg:1]: do we need this?

File: Observer/SalesOrderShipmentAfter.php (hunk 1 above)
Comment (LucianTuriacArnia) [reply Observer/SalesOrderShipmentAfter.php:170]: I will change the property name to  createInvoiceService

**Review by harli91 — APPROVED** [review]

**Pull request merged**